cas-server-support-osf/src/main/java/io/cos/cas/web/flow/OpenScienceFrameworkLoginHandler.java

```diff
@@ -106,17 +126,29 @@ public static OpenScienceFrameworkLoginContext fromJson(final String jsonString)
     }
 
     /**
-     * Decision making action that handles OSF login: osf, institution, campaign.
+     * Decision making action that handles OSF login.
      *
      * @param context The request context
-     * @return Event "osf", "institution" or "campaigns
+     * @return Event "osfDefaultLogin", "institutionLogin"
      */
     public Event beforeLogin(final RequestContext context) {
 
         final String serviceUrl = getEncodedServiceUrl(context);
         final boolean institutionLogin = isInstitutionLogin(context);
 
-        final OpenScienceFrameworkLoginContext osfLoginContext = new OpenScienceFrameworkLoginContext(serviceUrl, institutionLogin);
+        final String jsonLoginContext = (String) context.getFlowScope().get("jsonLoginContext");
+        final OpenScienceFrameworkLoginContext osfLoginContext;
+
+        if (jsonLoginContext == null) {
+            // Create a new Login Context with service URL and institution flag
+            osfLoginContext = new OpenScienceFrameworkLoginContext(serviceUrl, institutionLogin);
+        } else {
+            // Use current Login Context which contains information of Authentication Exceptions
```

Comment:
Need to retrieve the `handleErrorName` from the current `loginContext` when submission fails.